plugins/org.jboss.tools.teiid.reddeer/src/org/jboss/tools/teiid/reddeer/view/ConnectionView.java

```diff
@@ -0,0 +1,147 @@
+                return false;
+            } else {
+                throw ex; // another problem
+            }
+        }
+        return true;
+    }
+    
+    public void createConnectionProfile() {
+        List<String> path = new ArrayList<String>();
+        path.add("Local Profiles");
+        new PushButton(this, new WithTooltipTextMatcher("Create Connection Profile")).click();
+        new DefaultShell("New Connection Profile"); // TODO create class for it?
+    }
+
+    public void createDataSource() {
+        new DefaultTreeItem("Deployed").select();
+        new PushButton(this, new WithTooltipTextMatcher("Create Data Source")).click();
+        new DefaultShell("Create DataSource"); // TODO create class for it?
+    }
+
+    public void deleteConnectionProfile(String... inputPath) {
+        refreshView();
+        List<String> path = new ArrayList<String>();
+        path.add("Local Profiles");
+        path.addAll(Arrays.asList(inputPath));
+        new DefaultTreeItem(path.toArray(new String[0])).select();
+        new PushButton(this, new WithTooltipTextMatcher("Delete")).click();
+        DefaultShell deleteDialog = new DefaultShell("Delete confirmation");
+        new YesButton(deleteDialog).click();
+        refreshView();
+    }
+
+    public void deleteDataSource(String name) {
+        refreshView();
+        new DefaultTreeItem("Deployed", name).select();
+        new PushButton(this, new WithTooltipTextMatcher("Delete")).click();
+        DefaultShell deleteDialog = new DefaultShell("Delete DataSource");
+        new YesButton(deleteDialog).click();
+        new WaitWhile(new ShellIsActive("Progress Information"));
+        refreshView();
+    }
+
+    public void editConnectionProfile(String... inputPath) {
+        refreshView();
+        List<String> path = new ArrayList<String>();
+        path.add("Local Profiles");
+        path.addAll(Arrays.asList(inputPath));
+        new DefaultTreeItem(path.toArray(new String[0])).select();
+        new PushButton(this, new WithTooltipTextMatcher("Edit")).click();
+        new DefaultShell("Properties for " + path.get(path.size()-1));
+    }
+
+    public void editDataSource(String name) {
+        refreshView();
+        new DefaultTreeItem("Deployed", name).select();
+        new PushButton(this, new WithTooltipTextMatcher("Edit")).click();
+        new DefaultShell("Edit DataSource"); // TODO create class for it?
+    }
+
+    public void refreshView() {
+        new PushButton(this, new WithTooltipTextMatcher("Refresh profiles and deployed data sources")).click();
+    }
+
+    public void changeDefaultServer(String nameNewServer) {
+        new DefaultHyperlink(new WithTooltipTextMatcher("Change Default Server;"));
+        if (new ShellIsActive("Default server unchanged").test()) {
+            throw new Error("Only 1 server is configured");
+        }
+        if (new ShellIsActive("Server Selection").test()) {
```

Comment:
better usage is as follows
```
if (new ShellIsAvailable("Server Selection").test()) {
    shell = new DefaultShell("Server Selection");
    new DefaultCombo(shell).setSelection(...);
    new OkButton(shell).click();
}
```
the above code brings better stability to your tests. If we wanted to be more precise we could use
```
condition = new WaitUntil(new ShellIsAvailable("Server Selection"), TimePeriod.DEFAULT, false); // notice the 'false'
if (condition.getResult() != null) {
    new DefaultCombo(condition.getResult()).setSelection(...);
    new OkButton(condition.getResult()).click();
}
```
but this will take a lot of time.